src/app/views/forgotten-password/setForgottenPasswordController.jsx

```diff
@@ -11,12 +15,115 @@ export class SetForgottenPasswordController extends Component {
     constructor(props) {
         super(props);
         this.state = {
-            hasSubmitted: false
+            hasSubmitted: false,
+            isSubmitting: false,
+            passwordIsValid: false,
+            password: "",
+            showInputError: false
         };
+        this.onSubmit = this.onSubmit.bind(this);
+        this.validityCheck = this.validityCheck.bind(this);
+    }
+
+    onSubmit(event) {
+        event.preventDefault();
+        if (!this.state.passwordIsValid) {
+            this.setState({ showInputError: true });
+            return;
+        }
+        let verificationID = new URLSearchParams(location.search).get("vid");
+        let userID = new URLSearchParams(location.search).get("uid");
+        // UID is sent in Email field, for speed of first delivery it was decided UID is to be used instead of email.
+        // Eventually the backend should be changed to use the endpoint /users/{uid}/password or at least change the
+        // email field to be uid and then (TODO) this can be updated
+        const requestBody = {
```

Comment:
can you maybe move this out of here I think it is part of the request not submit